include/deal.II/lac/petsc_ts.h

```diff
@@ -0,0 +1,706 @@
+          AssertTS(KSPSetType(ksp, KSPPREONLY));
+          AssertTS(KSPSetPC(ksp, precond.get_pc()));
+        }
+
+      // Attach user monitoring routine
+      if (monitor)
+        AssertTS(TSMonitorSet(ts, _dealii_ts_monitor_, nullptr, nullptr));
+
+      AssertTS(TSSetFromOptions(ts));
+
+      AssertTS(TSSolve(ts, nullptr));
+
+      PetscInt nt;
+      AssertTS(TSGetStepNumber(ts, &nt));
+      return nt;
+    }
+
+    /**
+     * Integrate the differential-algebraic equations.
+     * Here we also pass the matrices to handle Jacobians.
+     */
+    unsigned int
+    solve(VectorType &solution, PMatrixType &P)
+    {
+      reinit_matrices(P);
+      return solve(solution);
+    }
+
+    /**
+     * Integrate the differential-algebraic equations.
+     * Here we also pass the matrices to handle Jacobians.
+     */
+    unsigned int
+    solve(VectorType &solution, AMatrixType &A, PMatrixType &P)
+    {
+      reinit_matrices(A, P);
+      return solve(solution);
+    }
+
+    /**
+     * Compute implicit residual $F(t, y, \dot y)$.
+     */
+    std::function<int(const double      t,
+                      const VectorType &y,
+                      const VectorType &y_dot,
+                      VectorType &      res)>
+      implicit_function;
+
+    /**
+     * Compute implicit Jacobian
+     * $\dfrac{\partial F}{\partial y} + \alpha \dfrac{\partial F}{\partial \dot
+     * y}$
+     */
+    std::function<int(const double      t,
+                      const VectorType &y,
+                      const VectorType &y_dot,
+                      const double      alpha,
+                      AMatrixType &     A,
+                      PMatrixType &     P)>
+      implicit_jacobian;
+
+    /**
+     * Compute explicit residual $G(t, y)$.
+     */
+    std::function<int(const double t, const VectorType &y, VectorType &res)>
+      explicit_function;
+
+    /**
+     * Compute explicit Jacobian $\dfrac{\partial G}{\partial y}$.
+     */
+    std::function<
+      int(const double t, const VectorType &y, AMatrixType &A, PMatrixType &P)>
+      explicit_jacobian;
+
+    /**
+     * Process solution. This function is called by TimeStepper at the beginning
+     * of each time step.
+     */
+    std::function<
+      int(const double t, const VectorType &y, const unsigned int step_number)>
+      monitor;
+
+    /**
+     * Setup Jacobian without matrices. This callback gives full control to
+     * users to setup the linearized equations
+     * $\dfrac{\partial F}{\partial y} + \alpha \dfrac{\partial F}{\partial \dot
+     * y}$
+     */
+    std::function<int(const double      t,
+                      const VectorType &y,
+                      const VectorType &ydot,
+                      const double      shift)>
+      setup_jacobian;
+
+    /**
+     * Solution of the Jacobian system setup with setup_jacobian
+     */
+    std::function<int(const VectorType &src, VectorType &dst)>
+      solve_for_jacobian_system;
```

Comment:
Maybe
```suggestion
    std::function<int(const VectorType &src, VectorType &dst)>
      solve_with_jacobian;
```
?

For all these names, I recognize now that there is precedent in the ARKODE and IDA interfaces. Could we use the same names as used there?